src/main/java/gyro/google/storage/BucketAccessControlResource.java

```diff
@@ -0,0 +1,292 @@
+     * Unique ID for the resource.
+     */
+    @Id
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+    
+    /**
+     * The link to this access-control entry.
+     */
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+    
+    /**
+     * The email address associated with the entity.
+     */
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    /**
+     * The ID for the entity.
+     */
+    public String getEntityId() {
+        return entityId;
+    }
+
+    public void setEntityId(String entityId) {
+        this.entityId = entityId;
+    }
+
+    /**
+     * The project team configuration associated with the entity.
+     */
+    public BucketAccessControlProjectTeam getProjectTeam() {
+        return projectTeam;
+    }
+
+    public void setProjectTeam(BucketAccessControlProjectTeam projectTeam) {
+        this.projectTeam = projectTeam;
+    }
+
+    @Override
+    public boolean refresh() {
+        Storage storage = createClient(Storage.class);
+
+        try {
+            BucketAccessControl acl = storage.bucketAccessControls()
+                    .get(getBucket().getName(), getEntity())
+                    .setUserProject(getUserProject())
+                    .execute();
+
+            if (acl == null) {
+                return false;
+            }
+
+            copyFrom(acl);
+
+            return true;
+        } catch (GoogleJsonResponseException e) {
+            if (e.getDetails().getCode() == 404) {
+                return false;
+            } else {
+                throw new GyroException(e.getDetails().getMessage());
+            }
+        } catch (IOException e) {
+            throw new GyroException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void create(GyroUI ui, State state) throws Exception {
+        Storage storage = createClient(Storage.class);
+
+        BucketAccessControl acl = new BucketAccessControl();
+        acl.setBucket(getBucket().getName());
+        acl.setEntity(getEntity());
+        acl.setRole(getRole());
+
+        try {
+            copyFrom(storage.bucketAccessControls().insert(getBucket().getName(), acl).execute());
+        } catch (GoogleJsonResponseException e) {
+            throw new GyroException(e.getDetails().getMessage());
+        } catch (Exception e) {
+            throw new GyroException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        Storage storage = createClient(Storage.class);
+        BucketAccessControl acl = new BucketAccessControl();
+
+        if (changedFieldNames.contains("bucket")) {
+            acl.setRole(getBucket().getName());
```

Comment:
As `bucket` is not a required field, if its removed while doing an update, this might result in NPE.

Reply:
The bucket field should be required. Updated.